src/lib/utils/contributors.ts

```diff
@@ -0,0 +1,47 @@
+import { join } from "path"
+
+import type { CommitHistory, FileContributor, Lang, Layout } from "@/lib/types"
+
+import { DEFAULT_LOCALE } from "@/lib/constants"
+
+import {
+  convertToFileContributorFromCrowdin,
+  getCrowdinContributors,
+} from "./crowdin"
+import { getLastModifiedDate } from "./gh"
+
+import { fetchAndCacheGitContributors } from "@/lib/api/fetchGitHistory"
+
+export const getFileContributorInfo = async (
+  mdDir: string,
+  mdPath: string,
+  slug: string,
```

Comment:
Minor: I see too many args for this function. For example, `mdDir` and `mdPath` are derived from `slug`. Perhaps we could abstract that logic (how to calc mdDir and mdPath) somewhere in the utils and reduce the noise form this function.

IMO, ideally, this function should only require 2 args: slug and locale (the cache is arguable but ok).